### ISSUE TITLE
Check SegmentationPlatformFeature and OptimizationHints new features

### ISSUE THREAD

**Issue by uazo** [bromite/bromite#1632]
### Is your feature request related to privacy?

Yes

### Is there a patch available for this feature somewhere?

No

### Describe the solution you would like

Check `kOptimizationHints` (enabled by default) and `kSegmentationPlatformFeature` (enabled by default from 2021-11-16)

```
// Feature flag for segmentation platform dummy model that is used for
// experimental models and data collection.

// Enables the syncing of the Optimization Hints component, which provides
// hints for what optimizations can be applied on a page load.

// Enables fetching from a remote Optimization Guide Service.
```

### Describe alternatives you have considered

None, for now

**Comment by csagan5** [comment]
Fetching from any "Optimization Guide Service" is impossible so I wonder how can this be currently working.

**Comment by uazo** [comment]
ungoogled did 

https://github.com/Ahrotahn/ungoogled-chromium/blob/master/patches/extra/ungoogled-chromium/disable-remote-optimization-guide.patch

but I still haven't seen if the other flags above should be turned off as well

**Comment by uazo** [comment]
code checked, it seems to me it works like this.
at each page load, and at each user click, blink examines the anchors and sends them to the browser.
the browser, using Tensor Flow (lite) applies models retrieved from google servers (specific to that url) and checks which hints to apply (https://source.chromium.org/chromium/chromium/src/+/main:components/optimization_guide/proto/hints.proto;l=103;bpv=1;bpt=1).
The same logic seems to be used, for example, to generate topics related to navigation which are then recorded in the history and turned to google servers via sync, I imagine that this information is used to produce the values ​​necessary for floc (but obviously I don't know with certainty).
TFLite models are recovered in various ways:
- via component update
- by a request to https://optimizationguide-pa.googleapis.com based on the links sent by blink
- periodically through the download manager

hints can also be received via android notifications

These urls need the google api key, but the request seems to start even if the key is empty. Some information returns to google via the ukm uploader.
The download is not active in incognito mode, but the navigation uses the models eventually downloaded from the normal navigation.

So we can remove everything, currently it cannot be used.

Now I look at what SegmentationPlatformFeature is.

**Comment by uazo** [comment]
```
SegmentationPlatformFeature
An internal component of chrome that uses UMA signals and ML models to customize browser 
experience for certain segments of users.
```
It applies tensorflow models to the user's habits of use on the ui, possibly modifying it (activate and deactivate buttons).
Here too, since the models were downloaded from google's servers, through the `OptimizationHints`, the code would not produce any effect.
In any case, it bothers me personally that there is something that possibly analyzes my behavior, leaving a trace on disk, so I would deactivate it.

**Comment by uazo** [comment]
[here](https://github.com/uazo/bromite/blob/master-v97/build/patches/Disable-all-predictors-code.patch) my version to disable those features

**Comment by csagan5** [comment]
> In any case, it bothers me personally that there is something that possibly analyzes my behavior, leaving a trace on disk, so I would deactivate it.

I think the bigger risk is that this data is sent to online servers.

I have added your patch to v99.

**Comment by csagan5** [comment]
Fixed in `99.0.4844.55`.
